### DISCUSSION TITLE
Refactoring: cleaning up the debt

### DISCUSSION THREAD

**Discussion by NicolasDorier** [btcpayserver/btcpayserver#2352] [General]
Opening the discussion to gather information about pain points that we should cleanup in code in a later release.

@Kukks @dennisreimann @rockstardev @bolatovumar

**Comment by NicolasDorier** [comment]
* [ ] Refactor the `IsHotWallet`, into a `PermissionService`. So we don't have to always DI the settings service and the authorization manager for all classes depending on IsHotWallet

**Reply by dennisreimann** [reply]
There are also some things that the [CssThemeManager](https://github.com/btcpayserver/btcpayserver/blob/master/BTCPayServer/HostedServices/CssThemeManager.cs#L66) takes care of, which don't belong there – we should refactor that out or rename the class.

**Reply by Kukks** [reply]
> There are also some things that the [CssThemeManager](https://github.com/btcpayserver/btcpayserver/blob/master/BTCPayServer/HostedServices/CssThemeManager.cs#L66) takes care of, which don't belong there – we should refactor that out or rename the class.

Yes, CssThemeManager should be removed entirely actually. It is acting as a caching proxy to a specific setting saved via `SettingsRepository`. We should build the caching system directly into that service and make it generic so that it  caches all settings (and updates the cache on save)

**Reply by Kukks** [reply]
> * [ ]  Refactor the `IsHotWallet`, into a `PermissionService`. So we don't have to always DI the settings service and the authorization manager for all classes depending on IsHotWallet

We can instead just convert it into policies and let the authorizationservice handle it itself? We can have `CanViewSeed`, `CanUseHotWalletToSpend`, etc

**Reply by NicolasDorier** [reply]
This is what PermissionService would do, just call authorizationService. But sometimes we need to do more specific check depending on the parameters of the route and the policy settings of the store, and whether it is cookie auth or greenfield.
